text/3750-cfg-os-version-min.md

```diff
@@ -0,0 +1,194 @@
+Because of this diversity in version strings each platform will be responsible for defining an `is_gte` comparison function for each key they support.
+Alternatively they can use one of the pre-defined functions.
+
+## Linting
+[linting]: #linting
+
+By default `os_version_min` will be linted by `check_cfg` in a similar way to `target_os`.
+That is, all valid values for `target_os` will be accepted as valid keys for `os_version_min` on all platforms.
+The list of additional keys supported by the target will be consulted, which will then be allowed on a per-target basis.
+
+## Future Compatibility
+[future-compatibility]: #future-compatibility
+
+The functions for parsing and comparing version strings may need to be updated whenever a new API is added, when the version format changes, or when new aliases need to be added.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Each supported platform will need to implement version string parsing logic (or re-use some provided defaults), maintain the logic in response to future changes, and update any version alias tables.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+The overall mechanism proposed here builds on other well established primitives in Rust such as `cfg`.
+A mechanism which tries to bridge cross-platform differences under one `min_target_api_version` predicate [was suggested](https://github.com/rust-lang/rfcs/blob/b0f94000a3ddbd159013e100e48cd887ba2a0b54/text/0000-min-target-api-version.md) but was rejected due to different platforms having divergent needs.
+
+For many platforms, the `target_os` name and the `os_version_min` name will be identical.
+Even platforms that have multiple possible `versions` relevant to the OS will still have one primary version.
+E.g. for `linux` the primary version would refer to the kernel with `libc` being a secondary OS library version.
+Therefore it would make sense for the primary target OS version to be a property of `target_os`.
+E.g.: `cfg(target_os("macos", min_version = "..."))`.
+This means we'd need a more general syntax for `libc` and potentially other versioned libraries where the target OS is ambiguous.
+
+# Prior art
+[prior-art]: #prior-art
+
+In C it's common to be able to target different versions based on a preprocessor macro.
+For example, on Windows `WINVER` can be used:
+
+```c
+// If the minimum version is at least Windows 10
+#if (WINVER >= _WIN32_WINNT_WIN10)
+// ...
+#endif
+```
+
+This RFC is a continuation of [RFC #3379](https://github.com/rust-lang/rfcs/pull/3379) more narrowly scoped to just `os_version_min`.
+That RFC was in turn an updated version of [this RFC draft](https://github.com/rust-lang/rfcs/pull/3036), with the changes reflecting conversations from the draft review process and [further Zulip discussion](https://rust-lang.zulipchat.com/#narrow/stream/213817-t-lang/topic/CFG.20OS.20Redux.20.28migrated.29/near/294738760).
+
+# Unresolved questions
```

Comment:
Another unresolved question to add: How should this work in Cargo `[target.'cfg(os_version_min(...))'.dependencies]` sections?